addons/account/wizard/account_accrual_accounting.py

```diff
@@ -0,0 +1,121 @@
+            raise UserError(_('All accounts on the lines must be from the same type.'))
+        rec['account_type'] = active_move_line_ids[0].account_id.user_type_id.internal_group
+        return rec
+
+    def amend_entries(self):
+        move_data = []
+        message_data = []
+        amend_data = []
+        for aml in self.active_move_line_ids:
+            reported_debit = aml.company_id.currency_id.round((self.percentage / 100) * aml.debit)
+            reported_credit = aml.company_id.currency_id.round((self.percentage / 100) * aml.credit)
+            move_data.append({
+                'date': self.date,
+                'ref': _('Accrued {type} for invoice: {name}').format(type=_('Revenue') if self.account_type == 'income' else _('Expense'), name=aml.move_id.name),
+                'line_ids': [
+                    (0, 0, {
+                        'debit': reported_debit,
+                        'credit': reported_credit,
+                        'account_id': aml.account_id.id,
+                        'name': aml.name,
+                        'partner_id': aml.partner_id.id,
+                    }),
+                    (0, 0, {
+                        'debit': reported_credit,
+                        'credit': reported_debit,
+                        'account_id': (self.revenue_accrual_account if self.account_type == 'income' else self.expense_accrual_account).id,
+                        'partner_id': aml.partner_id.id,
+                    }),
+                ],
+                'journal_id': self.journal_id.id,
+            })
+            # If percentage != 100, we need to split the original line into two lines
+            amend_data.append({
+                'line_ids': [
+                    (1, aml.id, {
+                        'debit': reported_debit,
+                        'credit': reported_credit,
+                        'account_id': (self.revenue_accrual_account if self.account_type == 'income' else self.expense_accrual_account).id,
+                    }),
+                    (0, 0, {
+                        'debit': aml.debit - reported_debit,
+                        'credit': aml.credit - reported_credit,
+                        'account_id': aml.account_id.id,
+                        'name': aml.name,
+                        'partner_id': aml.partner_id.id,
+                        'currency_id': aml.currency_id.id,
+                        'analytic_tag_ids': [(6, 0, aml.analytic_tag_ids.ids)],
+                        'analytic_account_id': aml.analytic_account_id.id,
+                        'tax_ids': [(6, 0, aml.tax_ids.ids)],
+                        'tag_ids': [(6, 0, aml.tag_ids.ids)],
+                    })
+                ]
+            })
+            message_data.append(_('Accrued {type} for invoice: <a href="/web#id={id}&model=account.move">{name}</a>').format(
```

Comment:
It works this way for the translations

Reply:
It could work but it's not the good way to deal with translations. Here is an example:
Suppose `Accrued Revenue` is `Produit de Régularisation` and `Accrued Expense` is `Charge à Payer`: you can't manage such translations with your way.

Reply:
I know the grammar of a few languages and in none of them will the order of words (adjective-noun, or noun-adjective) change depending on the noun.

Reply:
I could try a mathematical explanation if you want: `f(x) + f(y) != f(x + y)`

Reply:
You can't combine words are translations like you want as the translation of each word independently is not necessary the same as the translation of the full sentence

Reply:
`_('Accrued') + _('Revenue') != _('Accrued Revenue')`

Reply:
I agree with you that the purpose of the task (and thus it's implementation) is maybe flawed. And for this we should wait on the input of @avw. But really on this translation thing, I think you are completely out.

Your mathematical analogy is completely flawed. What you have is a function `f(type, id, name)` and the translator gives a new function `g(type, id, name)`. This is completely valid, and I'm not trying to add 2 functions or whatever. Even so, `_('{accrued}') + _('{revenue}') == _('{accrued} {revenue}')` if you have any notion in basic grammar.

Reply:
Your logic is stupid.

Reply:
Basic example: "Journal Items":
_('Journal') = "Journal"
_('items') = "éléments"
_('Journal Items') = "Ecritures comptables"

CQFD